lib/ansible/modules/cloud/amazon/ec2_elb_facts.py

```diff
@@ -201,20 +203,28 @@ def _get_elb_info(self, elb):
 
 
     def list_elbs(self):
-        elb_array = []
-
-        try:
-            all_elbs = self.connection.get_all_load_balancers()
-        except BotoServerError as err:
-            self.module.fail_json(msg = "%s: %s" % (err.error_code, err.error_message))
+        elb_array, token = [], None
 
-        if all_elbs:
-            if self.names:
-                for existing_lb in all_elbs:
-                    if existing_lb.name in self.names:
-                        elb_array.append(existing_lb)
+        while True:
+            try:
+                all_elbs = self.connection.get_all_load_balancers(marker=token)
+                token = all_elbs.next_token
+            except BotoServerError as err:
+                self.module.fail_json(msg = "%s: %s" % (err.error_code, err.error_message),
+                    exception=traceback.format_exc())
+
+            if all_elbs:
+                if self.names:
```

Comment:
Wouldn't it be better to use the names argument with `get_all_load_balancers` - as long as self.names defaults to None, this should work

```
all.elbs = self.connection.get_all_load_balancers(marker=token, load_balancer_names=self.names)
```

http://boto.cloudhackers.com/en/latest/ref/elb.html#boto.ec2.elb.ELBConnection.get_all_load_balancers